cmd/username.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/okteto/okteto/pkg/cmd/login"
+	"github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/cobra"
+)
+
+//Username returns the username of the authenticated user
+func Username() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "username",
+		Short: "Returns the username of the authenticated user",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			username := okteto.GetUsername()
+			if username != "" {
+				fmt.Println(username)
+				return nil
+			}
+			t, err := okteto.GetToken()
```

Comment:
why is it refreshing the token? if the user is not logged in I think it's better to ask them to log in again.

Reply:
it is because we were not saving the external-id. if the user is logged, but the external-id is not saved locally, we automatically do login to get the external-id